include/material/mohr_coulomb.h

```diff
@@ -0,0 +1,136 @@
+#ifndef MPM_MATERIAL_MOHR_COULOMB_H_
+#define MPM_MATERIAL_MOHR_COULOMB_H_
+
+#include <limits>
+
+#include <cmath>
+
+#include "Eigen/Dense"
+
+#include "material.h"
+
+namespace mpm {
+
+//! MohrCoulomb class
+//! \brief Mohr Coulomb material model
+//! \details Mohr Coulomb material model with softening
+//! \tparam Tdim Dimension
+template <unsigned Tdim>
+class MohrCoulomb : public Material<Tdim> {
+ public:
+  //! Define a vector of 6 dof
+  using Vector6d = Eigen::Matrix<double, 6, 1>;
+  //! Define a Matrix of 6 x 6
+  using Matrix6x6 = Eigen::Matrix<double, 6, 6>;
+
+  //! Failure state
+  enum FailureState { Elastic = 0, Tensile = 1, Shear = 2 };
```

Comment:
```suggestion
  enum class FailureState { Elastic, Tensile, Shear};
```

Reply:
Is it necessary to create a new namespace, or just put it in the "mpm"

Reply:
Yes, namespace is necessary.

Reply:
I have create mohrcoulomb namespace.